tests/xtimer_usleep/main.c

```diff
@@ -14,35 +15,45 @@
  * @brief       xtimer_usleep test application
  *
  * @author      Francisco Acosta <francisco.acosta@inria.fr>
+ * @author      Martine Lenders <m.lenders@fu-berlin.de>
  * @}
  */
 
+#include <inttypes.h>
 #include <stdio.h>
 
 #include "xtimer.h"
 #include "timex.h"
 
-#define ONE_SEC_SLEEP     (1 * US_PER_SEC)
-#define FIVE_SEC_SLEEP    (5 * US_PER_SEC)
-#define TEN_SEC_SLEEP     (10 * US_PER_SEC)
+#define RUNS                (5U)
+#define SLEEP_TIMES_NUMOF   (sizeof(sleep_times) / sizeof(sleep_times[0]))
+
+static const uint32_t sleep_times[] = { 10000, 50000, 100000 };
 
 int main(void)
 {
-    int t = 9;
-
-    printf("This test will print \"Slept for X sec...\" every 1, 5 and 10 seconds.\n");
-    printf("\n<======== If using pyterm, this is the time when started.\n\n");
-
-    while (t) {
-        xtimer_usleep(ONE_SEC_SLEEP);
-        printf("Slept for 1 sec...\n");
-        xtimer_usleep(FIVE_SEC_SLEEP);
-        printf("Slept for 5 sec...\n");
-        xtimer_usleep(TEN_SEC_SLEEP);
-        printf("Slept for 10 sec...\n");
-        t--;
+    uint32_t start_test, testtime;
+
+    printf("Running test %u times with %u distinct sleep times\n", RUNS,
+           SLEEP_TIMES_NUMOF);
+    puts("Please hit any key and then ENTER to continue");
+    getchar();
+    start_test = xtimer_now_usec();
+    for (unsigned m = 0; m < RUNS; m++) {
+        for (unsigned n = 0;
+             n < sizeof(sleep_times) / sizeof(sleep_times[0]);
+             n++) {
+            uint32_t diff, start;
+            printf("Sleeping for %" PRIu32 " us\n", sleep_times[n]);
+            start = xtimer_now_usec();
+            xtimer_usleep(sleep_times[n]);
+            diff = xtimer_now_usec() - start;
+            printf("Slept for %" PRIu32 " us\n", diff);
```

Comment:
You could get rid of one `printf` here -> `printf("Slept for xx us (target: xx us)");`. Keeping the number of printfs low while doing time measurements helps to reduce timing overheads on the overall time measurement...